paymentsheet/src/main/java/com/stripe/android/paymentsheet/repositories/CustomerRepository.kt

```diff
@@ -33,4 +33,13 @@ internal interface CustomerRepository {
         customerConfig: PaymentSheet.CustomerConfiguration,
         paymentMethodId: String
     ): PaymentMethod?
+
+    /**
+     * Attach a payment method to the Customer and return the modified [PaymentMethod].
+     * Silently handle failures by returning null.
+     */
+    suspend fun attachPaymentMethod(
+        customerConfig: PaymentSheet.CustomerConfiguration,
+        paymentMethodId: String
+    ): PaymentMethod?
```

Comment:
Can this return a `Result<PaymentMethod>` instead?

Reply:
The rest of the methods don't return a Result and also silently returns null or empty list if it fails. Should we do this refactor for all of the methods here?

Reply:
We should start with this one and eventually transition the others too.

Reply:
I feel we should do the refactor of the rest of customer repository separately. If it's not a ton of work, we can add Result here now, but we shouldn't refactor the rest.